File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    int getListLength(ListNode head) {
        int len = 0;
        ListNode curr = head;
        while(curr != null) {
            len++;
            curr = curr.next;
        }
        return len;
    }
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int len = getListLength(head);
        int targetInd = len - n;
        if(targetInd == 0) return head.next;
        ListNode prev = head;
        ListNode curr = head.next;
        int i = 0;
        while(i < targetInd-1) {
            prev = curr;
            curr = curr.next;
            i++;
        }
        if(curr == null) {
            prev.next = null;
        } else {
            prev.next = curr.next;
            curr.next = null;
        }
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
    let len = 0
    let curr = head
    while(curr != null) {
        curr = curr.next
        len++
    }
    if(len === n) {
        return head.next
    }
    // console.log("len: ", len)
    curr = head
    let i=0
    while(curr != null && (len-i !== n+1)) {
        // console.log("curr: ", curr)
        curr = curr.next
        i++
    }
    const next = curr.next
    curr.next = next.next
    next.next = null
    return head
};